### PR TITLE
Make Providers more easily extendable

### PR DESCRIPTION
**Pull request by pannal**
by adding overridable Provider.subtitle_class

### CODE REVIEW

**Comment by Diaoul** [comment]
Good idea!
LGTM

File: subliminal/providers/legendastv.py [hunk 1]
```diff
@@ -276,9 +276,9 @@ def get_archives(self, title_id, language_code):
             soup = ParserBeautifulSoup(r.content, ['lxml', 'html.parser'])
             for archive_soup in soup.select('div.list_element > article > div'):
                 # create archive
-                archive = LegendasTVArchive(archive_soup.a['href'].split('/')[2], archive_soup.a.text,
+                archive = LegendasTVArchive(archive_soup.p.a['href'].split('/')[2], archive_soup.p.a.text,
```
Comment (fernandog) [inline subliminal/providers/legendastv.py]: @pannal this fix doesn't work for non-donors. I have this fix that works for both and have pytest/cassetes:

https://github.com/Diaoul/subliminal/pull/762/commits/02829e7d5edd6d55fc70ed69cd825efe361531d6

Comment (pannal) [inline subliminal/providers/legendastv.py]: Ah, nice. Merging that.

**Comment by pannal** [comment]
Whoops, this PR has gone out of hand. Sorry.

**Pull request closed without merging**